homework/tests/example.spec.js

```diff
@@ -14,18 +15,184 @@ const data = [
   '9'
 ]
 
-
 data.forEach(version => {
   test.describe(version + ': Add', () => {
     test('Concatenating 2 and 3 results in 23', async ({ page }) => {
-      await page.goto('https://testsheepnz.github.io/BasicCalculator');
-      await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
-      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-      await page.locator('#calculateButton').click();
-  
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+      let calculatorPage = new CalculatorPage(page);
+
+      await calculatorPage.navigate();
+      await calculatorPage.getVersion(version);
+      await calculatorPage.click('#number1Field');
+      await calculatorPage.fillField('#number1Field', '2');
+      await calculatorPage.click('#number2Field');
+      await calculatorPage.fillField('#number2Field', '3');
+      await calculatorPage.selectOption('select[name="selectOperation"]', '4');
+      await calculatorPage.click('#calculateButton');
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('23');
     });
   });
-});
\ No newline at end of file
+
+  test.describe(version + ': Add', () => {
```

Comment:
do you need a separate describe block for each test, are you not trying to group them in any way?

Reply:
In this case, I don't know any other group method. Any suggestions, on what could be changed?

Reply:
what I meant is that you can have more than one test within a describe block. So for example here, you have a few tests that try to add different values and expect different results.  All those tests for add operation could be grouped inside one describe block instead of having a separate one each.